src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -70,7 +69,27 @@ public synchronized void delete(Long id) {
 
     @Override
     public synchronized List<Product> sort(List<Product> unsortedProducts, String productField, String order){
-        return new ArrayList<>();
+        if (productField != null && !productField.isEmpty()) {
+            return unsortedProducts.stream()
+                    .sorted(getComparator(productField, order))
+                    .collect(Collectors.toList());
+        } else {
+            return unsortedProducts;
+        }
+    }
+
+    private Comparator<Product> getComparator(String productField, String order) {
+        comparators = new HashMap<>();
+        comparators.put("price", Comparator.comparing(Product::getPrice));
```

Comment:
Do we create and populate comparators map each time we sort/call getComparator()?

Please move code to create and populate the map in another method and call it once.
getComparator should then use the previously populated map to extract appropriate value.